controller/src/beerocks/master/tasks/association_handling_task.cpp

```diff
@@ -132,37 +132,11 @@ void association_handling_task::work()
         TASK_LOG(DEBUG) << "started association_handling_task, looking for beacon measurement "
                            "capabilities on "
                         << sta_mac;
-        std::string bssid     = database.get_node_parent(sta_mac);
-        std::string radio_mac = database.get_node_parent_radio(bssid);
-        auto agent_mac        = database.get_node_parent_ire(bssid);
-
-        auto measurement_request = message_com::create_vs_message<
-            beerocks_message::cACTION_CONTROL_CLIENT_BEACON_11K_REQUEST>(cmdu_tx, id);
-
-        if (measurement_request == nullptr) {
-            LOG(ERROR) << "Failed building ACTION_CONTROL_CLIENT_BEACON_11K_REQUEST message!";
+        if (!son::son_actions::send_beacon_metrics_query_msg(cmdu_tx, database,
+                                                             tlvf::mac_from_string(sta_mac))) {
+            LOG(ERROR) << "Failed sending BEACON_METRICS_QUERY_MESSAGE for sta: " << sta_mac;
             return;
         }
-        measurement_request->params().measurement_mode =
-            beerocks::MEASURE_MODE_ACTIVE; // son::eMeasurementMode11K "passive"/"active"/"table"
-        measurement_request->params().channel = database.get_node_channel(bssid);
-        measurement_request->params().op_class =
-            database.get_hostap_operating_class(tlvf::mac_from_string(bssid));
-        measurement_request->params().rand_ival = beerocks::
-            BEACON_MEASURE_DEFAULT_RANDOMIZATION_INTERVAL; // random interval - specifies the upper bound of the random delay to be used prior to making the measurement, expressed in units of TUs [=1024usec]
-        measurement_request->params().duration = beerocks::
-            BEACON_MEASURE_DEFAULT_ACTIVE_DURATION; // measurement duration, expressed in units of TUs [=1024usec]
-        measurement_request->params().sta_mac = tlvf::mac_from_string(sta_mac);
-        measurement_request->params().bssid   = tlvf::mac_from_string(bssid);
-        //measurement_request.params.use_optional_ssid = true;
-        measurement_request->params().expected_reports_count = 1;
```

Comment:
There are a lot of parameters that we use to send and now we don't. How do we make up for it?
Did you fixate these parameters to hard-coded value on the monitor side?
Please add a description of it on the commit message.